src/core/net/ip6.cpp

```diff
@@ -1103,26 +1096,18 @@ otError Ip6::ProcessReceiveCallback(const Message &    aMessage,
     }
 
     // make a copy of the datagram to pass to host
-    VerifyOrExit((messageCopy = aMessage.Clone()) != nullptr, error = OT_ERROR_NO_BUFS);
-    IgnoreError(RemoveMplOption(*messageCopy));
-    mReceiveIp6DatagramCallback(messageCopy, mReceiveIp6DatagramCallbackContext);
-
-exit:
+    messageCopy = aMessage.Clone();
 
-    switch (error)
+    if (messageCopy == nullptr)
     {
-    case OT_ERROR_NO_BUFS:
         otLogWarnIp6("Failed to pass up message (len: %d) to host - out of message buffer.", aMessage.GetLength());
-        break;
-
-    case OT_ERROR_DROP:
-        otLogNoteIp6("Dropping message (len: %d) from local host since next hop is the host.", aMessage.GetLength());
-        break;
```

Comment:
Removed this log.
- It is not longer valid (the error returned in this situation is actually `OT_ERROR_NO_ROUTE` and not `DROP`)
- I think I added this log line in early development days of OT. This was helpful back then to detect a bunch of failure situations (e.g., when another node is offline and a message to it is being bounced back to host). We now have a lot more useful logs that provide more details in different situations, so I don't think this log adds any value.